src/inference/dev_api/openvino/runtime/icore.hpp

```diff
@@ -0,0 +1,251 @@
+     */
+    virtual ov::SoPtr<InferenceEngine::IExecutableNetworkInternal> compile_model(const std::string& model_str,
+                                                                                 const ov::Tensor& weights,
+                                                                                 const std::string& device_name,
+                                                                                 const ov::AnyMap& config) const = 0;
+
+    /**
+     * @brief Creates an executable network from a previously exported network
+     * @param model model stream
+     * @param device_name Name of device load executable network on
+     * @param config Optional map of pairs: (config parameter name, config parameter value) relevant only for this load
+     * operation*
+     * @return An executable network reference
+     */
+    virtual ov::SoPtr<InferenceEngine::IExecutableNetworkInternal>
+    import_model(std::istream& model, const std::string& device_name = {}, const ov::AnyMap& config = {}) const = 0;
+
+    /**
+     * @brief Query device if it supports specified network with specified configuration
+     *
+     * @param model OpenVINO Model
+     * @param device_name A name of a device to query
+     * @param config Optional map of pairs: (config parameter name, config parameter value)
+     * @return An object containing a map of pairs a layer name -> a device name supporting this layer.
+     */
+    virtual ov::SupportedOpsMap query_model(const std::shared_ptr<const ov::Model>& model,
+                                            const std::string& device_name,
+                                            const ov::AnyMap& config) const = 0;
+
+    /**
+     * @brief Returns devices available for neural networks inference
+     *
+     * @return A vector of devices. The devices are returned as { CPU, GPU.0, GPU.1, MYRIAD }
+     * If there more than one device of specific type, they are enumerated with .# suffix.
+     */
+    virtual std::vector<std::string> get_available_devices() const = 0;
+
+    /**
+     * @brief Checks whether device supports Export & Import functionality of network
+     *
+     * @param device_name - A name of a device to get a metric value.
+     * @return True if device has IMPORT_EXPORT_SUPPORT metric in SUPPORTED_METRICS and
+     * this metric returns 'true', False otherwise.
+     */
+    virtual bool device_supports_import_export(const std::string& device_name) const = 0;
+
+    /**
+     * @brief Create a new shared context object on specified accelerator device
+     * using specified plugin-specific low level device API parameters (device handle, pointer, etc.)
+     * @param device_name Name of a device to create new shared context on.
+     * @param params Map of device-specific shared context parameters.
+     * @return A shared pointer to a created remote context.
+     */
+    virtual ov::RemoteContext create_context(const std::string& device_name, const AnyMap& args) const = 0;
```

Comment:
also, wondering whether any plugin need to call GetCore()->create_context()

Reply:
What is about auto plugin?

Reply:
from what remote handles AUTO will do it? AUTo does not have such information, it can be provided by external user only

Reply:
Here is an example: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/auto_batch/auto_batch.cpp#L742